In [9]:
from sklearn.datasets import fetch_california_housing

california_housing = fetch_california_housing(as_frame=True)

df = california_housing.frame
df.head()


MedInc  HouseAge  AveRooms  AveBedrms  Population  AveOccup  Latitude  \
0  8.3252      41.0  6.984127   1.023810       322.0  2.555556     37.88   
1  8.3014      21.0  6.238137   0.971880      2401.0  2.109842     37.86   
2  7.2574      52.0  8.288136   1.073446       496.0  2.802260     37.85   
3  5.6431      52.0  5.817352   1.073059       558.0  2.547945     37.85   
4  3.8462      52.0  6.281853   1.081081       565.0  2.181467     37.85   

   Longitude  MedHouseVal  
0    -122.23        4.526  
1    -122.22        3.585  
2    -122.24        3.521  
3    -122.25        3.413  
4    -122.25        3.422

In [10]:
#understand the data
df.info

<bound method DataFrame.info of        MedInc  HouseAge  AveRooms  AveBedrms  Population  AveOccup  Latitude  \
0      8.3252      41.0  6.984127   1.023810       322.0  2.555556     37.88   
1      8.3014      21.0  6.238137   0.971880      2401.0  2.109842     37.86   
2      7.2574      52.0  8.288136   1.073446       496.0  2.802260     37.85   
3      5.6431      52.0  5.817352   1.073059       558.0  2.547945     37.85   
4      3.8462      52.0  6.281853   1.081081       565.0  2.181467     37.85   
...       ...       ...       ...        ...         ...       ...       ...   
20635  1.5603      25.0  5.045455   1.133333       845.0  2.560606     39.48   
20636  2.5568      18.0  6.114035   1.315789       356.0  3.122807     39.49   
20637  1.7000      17.0  5.205543   1.120092      1007.0  2.325635     39.43   
20638  1.8672      18.0  5.329513   1.171920       741.0  2.123209     39.43   
20639  2.3886      16.0  5.254717   1.162264      1387.0  2.616981     39.37   

       

In [11]:
df.dtypes

MedInc         float64
HouseAge       float64
AveRooms       float64
AveBedrms      float64
Population     float64
AveOccup       float64
Latitude       float64
Longitude      float64
MedHouseVal    float64
dtype: object

In [12]:
#check if they have missing value or not
df.isnull().sum()

MedInc         0
HouseAge       0
AveRooms       0
AveBedrms      0
Population     0
AveOccup       0
Latitude       0
Longitude      0
MedHouseVal    0
dtype: int64

In [13]:
#rename column for better 
df.columns = [col.lower() for col in df.columns]
df.rename(columns = {
    'medinc':'median_income',
    'houseage': 'housing_age',
    'averooms': 'avg_rooms',
    'avebedrms': 'avg_bedrooms',
    'population': 'population',
    'aveoccup': 'avg_occupants',
    'latitude': 'latitude',
    'longitude': 'longitude',
    'medhouseval': 'median_house_value'
}, inplace=True)

In [14]:
df.head()

median_income  housing_age  avg_rooms  avg_bedrooms  population  \
0         8.3252         41.0   6.984127      1.023810       322.0   
1         8.3014         21.0   6.238137      0.971880      2401.0   
2         7.2574         52.0   8.288136      1.073446       496.0   
3         5.6431         52.0   5.817352      1.073059       558.0   
4         3.8462         52.0   6.281853      1.081081       565.0   

   avg_occupants  latitude  longitude  median_house_value  
0       2.555556     37.88    -122.23               4.526  
1       2.109842     37.86    -122.22               3.585  
2       2.802260     37.85    -122.24               3.521  
3       2.547945     37.85    -122.25               3.413  
4       2.181467     37.85    -122.25               3.422

In [15]:
#save as a clean csv file
df.to_csv('/Users/anhnguyendo/Documents/Python machine learning/Cali housing price project/California-house-pricing-prediction/data/processed/california_house_median_price_cleaned', index = False)

Column Name	Description

median_income - Median income of households in the block group (scaled, e.g., 3.5 = $35,000)

housing_age - Median age of houses in the block group (in years)

avg_rooms - Average number of rooms per household

avg_bedrooms - Average number of bedrooms per household

population - Total population in the block group

avg_occupants - Average number of occupants per household

latitude - Latitude of the block group’s geographic center

longitude - Longitude of the block group’s geographic center

median_house_value - Median house value in the block group (in $100,000 units — maxed at $500,000)